recipes/whatlies/meta.yaml

```diff
@@ -0,0 +1,116 @@
+{% set name = "whatlies" %}
+{% set version = "0.6.5" %}
+
+
+package:
+  name: {{ name|lower }}-meta
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/whatlies-{{ version }}.tar.gz
+  sha256: 0b97c2b5b0bbc5c4044278ea8b2d3222cbb125e1fbf7bd854bb7f09b557e13d7
+
+build:
+  number: 0
+  noarch: python
+requirements:
+# ref: https://gitter.im/conda-forge/conda-forge.github.io?at=6195afbdb5ba9e5a11c02791
+  host:
+    - python >=3.6,<3.8
+  run:
+    - python >=3.6,<3.8
+
+outputs:
+  - name: {{ name|lower }}
+    build:
+      noarch: python
+      script: {{ PYTHON }} -m pip install . -vv
+      run_exports:
+        - {{ pin_subpackage(name, min_pin='x.x', max_pin='x.x') }}
+    requirements:
+      host:
+        - pip
+        - python >=3.6,<3.8
+      run:
+        - python >=3.6,<3.8
+        # base_packages
+        - altair >=4.0.1
+        - bpemb >=0.3.0
+        - gensim ~=3.8.3
+        - matplotlib-base >=3.2.0
+        - scikit-learn >=0.24.0
+      run_constrained:
+        # umap_packages
+        - umap-learn >=0.4.0
+        # rasa_packages
+        - rasa >=2.7.0
+        # fasttext_packages
+        - fasttext ~=0.9.1
+        # spacy_packages
+        - spacy >=3.0.1
+        - spacy-lookups-data >=0.3.2
+        # s2v_packages (+= spacy_packages)
+        - sense2vec >=1.0.2
+        # tf_packages
+        - tensorflow >=2.3.0,<2.5
+        - tensorflow-text >=2.3.0
+        - tensorflow-hub >=0.8.0
+        # transformers_dep
+        - transformers >=4.3.0
+        # sentence_tfm_dep
+        - sentence-transformers >=0.3.8
```

Comment:
No, the normal package would be the one created with the `run-constraints`. So, I guess, it's okay to call it just `whatlies`. Right now I have only added a single layer of `extra` (`umap`). There will be quite a few to create the following:

```sh
- whatlies
- whatlies-umap
- whatlies-rasa
- whatlies-fasttext
- whatlies-spacy
- whatlies-sv2
- whatlies-tf
- whatlies-transformers
- whatlies-sentence-tfm
```

Basically, my plan is to emulate the whole list of `extras` in the `setup.py` of `whatlies` using the `outputs:sub-packages` construct. Because of my lack of experience here I am not sure if it is feasible or if it is a good/bad plan from maintainability perspective. What would you suggest?

Reply:
The package equivalent of `[all]` would be the one that actually installs all dependencies.

Reply:
> Because of my lack of experience here I am not sure if it is feasible or if it is a good/bad plan from maintainability perspective. What would you suggest?

If you have the time, it's a great approach! Most of the work is creating the package. Once that's done, it's not that different. The only maintainability difference is that you have more requirements to update but that's it.